### PR TITLE
feat(hub): add api to retrieve staker's total value and number of chips

### PR DESCRIPTION
**Pull request by kallydev**
http://localhost/nta/stakers/0x69982e017acc0fde3d1542205089a8d3eafcd1b7

```json
{
  "address": "0x69982e017acc0fde3d1542205089a8d3eafcd1b7",
  "total_staked_nodes": 1,
  "total_chips": 374,
  "total_staked_tokens": "199547980194376366760196",
  "current_staked_tokens": "199547980194376366760196"
}
```

### CODE REVIEW

File: docs/openapi.json [hunk 1]
```diff
@@ -513,6 +513,32 @@
                 }
             }
         },
+        "/nta/stakers/{staker_address}": {
+            "get": {
+                "summary": "Get staker's profile",
```
Comment (HenryQW) [inline docs/openapi.json:518]: Update the summary

File: docs/openapi.json [hunk 2]
```diff
@@ -513,6 +513,32 @@
...
+                "description": "Retrieve the total number and value of chips for a specific staker.",
```
Comment (HenryQW) [inline docs/openapi.json:519]: Add "at the time of staking."

File: docs/openapi.json [hunk 3]
```diff
@@ -3140,6 +3166,36 @@
                     }
                 }
             },
+            "StakerResponse": {
+                "description": "A successful response containing a list of chips based on the provided filters. Each chip includes detailed information such as its ID, node address, owner address, and metadata. The response also includes a cursor for pagination to fetch subsequent sets of results.",
```
Comment (HenryQW) [inline docs/openapi.json:3170]: Update the description

File: docs/openapi.json [hunk 4]
```diff
@@ -3140,6 +3166,36 @@
...
+                                    "example": "0x69982e017acc0fde3d1542205089a8d3eafcd1b7"
+                                },
+                                "chips": {
+                                    "type": "object",
+                                    "properties": {
+                                        "total_number": {
+                                            "type": "string",
+                                            "example": "0"
+                                        },
+                                        "total_value": {
```
Comment (HenryQW) [inline docs/openapi.json:3188]: Probably use "total_staked_tokens"

**Review by HenryQW — CHANGES_REQUESTED** [review]

File: docs/openapi.json [hunk 5]
```diff
@@ -3140,6 +3166,43 @@
                     }
                 }
             },
+            "StakerResponse": {
+                "description": "A successful response with the total and current number of staked tokens, as well as the staked number of nodes staked and chips owned by the specific staker.",
```
Comment (HenryQW) [inline docs/openapi.json:3170]: > as well as the staked number of nodes staked

Comment (HenryQW) [reply docs/openapi.json:3170]: what does it mean?

File: docs/openapi.json [hunk 6]
```diff
@@ -513,6 +513,32 @@
                 }
             }
         },
+        "/nta/stakers/{staker_address}": {
+            "get": {
+                "summary": "Get the number of nodes staked by the staker, as well as the amount of chips and tokens.",
+                "description": "Retrieve the total and current number of staked nodes, the number of chips owned, and the amount of tokens for the staker.",
```
Comment (HenryQW) [inline docs/openapi.json:519]: description does not match `StakerResponse`

File: docs/openapi.json [hunk 7]
```diff
@@ -3140,6 +3166,43 @@
...
+                            "type": "object",
+                            "properties": {
+                                "address": {
+                                    "type": "string",
+                                    "description": "The wallet address of the staker.",
+                                    "example": "0x69982e017acc0fde3d1542205089a8d3eafcd1b7"
+                                },
+                                "total_staked_tokens": {
+                                    "type": "string",
+                                    "description": "Total cost of chips owned by the staker.",
```
Comment (HenryQW) [inline docs/openapi.json:3183]: > Total cost of chips owned by the staker

this is really a confusing description, I suggest: Total number of tokens historically staked (including unstaked tokens).

File: docs/openapi.json [hunk 8]
```diff
@@ -3140,6 +3166,43 @@
...
+                                    "example": "199547980194376366760196"
+                                },
+                                "current_staked_nodes": {
+                                    "type": "integer",
+                                    "description": "Current number of nodes staked by the staker.",
```
Comment (HenryQW) [inline docs/openapi.json:3188]: I'm not sure why it's called "current_staked_nodes", it surely means "total_staked_nodes" right?

File: docs/openapi.json [hunk 9]
```diff
@@ -3140,6 +3166,43 @@
...
+                                    "example": 1
+                                },
+                                "current_owned_chips": {
```
Comment (HenryQW) [inline docs/openapi.json:3191]: Same for this, "current_owned_chips" is just "total_chips"

File: docs/openapi.json [hunk 10]
```diff
@@ -3140,6 +3166,43 @@
...
+                                    "type": "string",
+                                    "description": "Current number of chips owned by the staker.",
+                                    "example": "374"
+                                },
+                                "current_staked_tokens": {
+                                    "type": "string",
+                                    "description": "Current cost of chips owned by the staker.",
```
Comment (HenryQW) [inline docs/openapi.json:3198]: Again this is confusing, do not mix chips with staked tokens. Just use: total amount of tokens currently staked.

File: schema/stake_staker.go [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/shopspring/decimal"
+)
+
+type StakeStaker struct {
+	Address             common.Address  `json:"address"`
+	TotalStakedTokens   decimal.Decimal `json:"total_staked_tokens"`
+	CurrentStakedNodes  uint64          `json:"current_staked_nodes"`
+	CurrentOwnedChips   uint64          `json:"current_owned_chips"`
```
Comment (HenryQW) [inline schema/stake_staker.go:12]: use TotalChips, unless we are going to supply a historical count (I don't think so).

**Review by HenryQW — CHANGES_REQUESTED** [review]

**Review by HenryQW — APPROVED** [review]

**Review by brucexc — APPROVED** [review]

**Pull request merged**